Source/Core/Core/PowerPC/PPCAnalyst.cpp

```diff
@@ -476,30 +476,28 @@ void PPCAnalyzer::ReorderInstructions(u32 instructions, CodeOp *code)
 
 void PPCAnalyzer::SetInstructionStats(CodeBlock *block, CodeOp *code, GekkoOPInfo *opinfo, u32 index)
 {
-	code->wantsCR0 = false;
-	code->wantsCR1 = false;
-
 	if (opinfo->flags & FL_USE_FPU)
 		block->m_fpa->any = true;
 
 	if (opinfo->flags & FL_TIMER)
 		block->m_gpa->anyTimer = true;
 
-	// Does the instruction output CR0?
-	if (opinfo->flags & FL_RC_BIT)
-		code->outputCR0 = code->inst.hex & 1; //todo fix
-	else if ((opinfo->flags & FL_SET_CRn) && code->inst.CRFD == 0)
-		code->outputCR0 = true;
-	else
-		code->outputCR0 = (opinfo->flags & FL_SET_CR0) ? true : false;
-
-	// Does the instruction output CR1?
-	if (opinfo->flags & FL_RC_BIT_F)
-		code->outputCR1 = code->inst.hex & 1; //todo fix
-	else if ((opinfo->flags & FL_SET_CRn) && code->inst.CRFD == 1)
-		code->outputCR1 = true;
-	else
-		code->outputCR1 = (opinfo->flags & FL_SET_CR1) ? true : false;
+	code->outputCR = BitSet8(0);
+	if ((opinfo->flags & FL_RC_BIT && (code->inst.hex & 1)) || (opinfo->flags & FL_SET_CR0))
+		code->outputCR[0] = true;
+	if ((opinfo->flags & FL_RC_BIT_F && (code->inst.hex & 1)) || (opinfo->flags & FL_SET_CR1))
+		code->outputCR[1] = true;
+	// mtcrf: this one is special and can modify multiple CR regs
+	if (code->inst.OPCD == 31 && code->inst.SUBOP10 == 144)
```

Comment:
hmm, not a bad idea, though if we do that we have another ~dozen instructions we'll need to add helper functions for to be consistent, I think.